main/src/ca/uwaterloo/flix/api/FlixEvent.scala

```diff
@@ -39,7 +41,7 @@ object FlixEvent {
   /**
     * An event that is fired when new type constraints are collected for the given def symbol `sym`.
     */
-  case class NewConstraintsDef(sym: Symbol.DefnSym, tconstrs: List[TypeConstraint]) extends FlixEvent
+  case class NewConstraintsDef(defn: KindedAst.Def, infResult: InfResult, renv: RigidityEnv, tconstrs: List[TraitConstraint], tenv: TraitEnv, eqEnv: EqualityEnv, root: KindedAst.Root) extends FlixEvent
```

Comment:
@mlutze Are we at the point where this needs its own type?

Reply:
It would just be packed and unpacked again so i don't think there is much point

Reply:
ConstraintSystem? ConstraintContext?

Reply:
I want to hear Matts opinion on this and then we can probably merge.

Reply:
It feels a bit unrelated to this, but fair